roles/debian-hardening/tasks/main.yml

```diff
@@ -382,5 +381,22 @@
 - name: Update grub
   command: update-grub
 
+- name: Set grub password
+  template:
+    src: "01_password.j2"
+    dest: /etc/grub.d/01_password
+    group: "root"
+    owner: "root"
+    mode: 0755
+  when: not revert
+- name: Unset grub password
+  file:
+    state: absent
+    path: /etc/grub.d/01_password
+  when: revert
+
+- name: Update grub
```

Comment:
rather than re running update-grub, move Set grub password before the last update-grub

Reply:
I remember that we've seen the CI doesn't rollback the boot partition.
It would be a good idea now to run update-grub in the rollback playbook after reboot.
Please add a commit for that.

Reply:
I change the 01_password.j2 to override kernel boot parameter